### DISCUSSION TITLE
Should the policy consts dealing with fee rates, ie. DUST_RELAY_TX_FEE be of type `FeeRate`?

### DISCUSSION THREAD

**Discussion by junderw** [rust-bitcoin/rust-bitcoin#2259] [Refactoring]
I was thinking of 2 things:

1. We change the policy consts to match their logical types. (DUST_RELAY_TX_FEE -> FeeRate etc.)
2. We add additional consts onto the types themselves that mirror policy. (ie. add a `FeeRate::DUST_RELAY_TX_FEE`)

Thoughts?

**Comment by yancyribbens** [comment]
> We change the policy consts to match their logical types. (DUST_RELAY_TX_FEE -> FeeRate etc.)

My inital thought is either `DUST_RELAY_TX_FEE` is a bad name and should be `DUST_RELAY_TX_FEE_RATE` or it should be of type `Amount`..

**Reply by Kixunil** [reply]
Yes, the former, it absolutely shouldn't be `Amount` since it's not an absolute fee.

**Comment by yancyribbens** [comment]
> We add additional consts onto the types themselves that mirror policy. (ie. add a FeeRate::DUST_RELAY_TX_FEE.

I don't think @apoelstra likes conts in this project :)

**Reply by apoelstra** [reply]
It's fine if they're copies of mostly-arbitrary policy constants in Bitcoin Core. But not if they're "size of a 32-bit integer" constants :).

**Comment by Kixunil** [comment]
Yes, use the `FeeRate` type and either rename to `DUST_RELAY_TX_FEE_RATE` or simply `FeeRate::DUST_RELAY_TX`. We can use alias to make sure people find it in the doc.
